Sources/MapboxCoreNavigation/ReplayLocationManager.swift

```diff
@@ -88,50 +90,73 @@ open class ReplayLocationManager: NavigationLocationManager {
             synthesizedLocation = location
             delegate?.locationManager?(self, didUpdateLocations: [location])
             onTick?(currentIndex, location)
-            NSObject.cancelPreviousPerformRequests(withTarget: self, selector: #selector(tick), object: nil)
+            nextTickWorkItem?.cancel()
         }
 
         func scheduleNextTick(afterDelay delay: TimeInterval) {
-            perform(#selector(tick), with: nil, afterDelay: delay)
```

Comment:
Retains self. So, in case of looping, there is a retain cycle that can only be broken by explicitly stopping updates. GCD doesn't have this problem.